File: game/debug/DebugPanel.lua
-- [ts]: DebugPanel.ts
local ____lualib = require("lualib_bundle") -- 1
local __TS__Class = ____lualib.__TS__Class -- 1
local __TS__SetDescriptor = ____lualib.__TS__SetDescriptor -- 1
local __TS__New = ____lualib.__TS__New -- 1
local ____exports = {} -- 1
local ImGui = require("ImGui") -- 3
local ____Dora = require("Dora") -- 4
local Keyboard = ____Dora.Keyboard -- 4
local ____GameContext = require("game.core.GameContext") -- 5
local ctx = ____GameContext.ctx -- 5
____exports.DebugPanel = __TS__Class() -- 7
local DebugPanel = ____exports.DebugPanel -- 7
DebugPanel.name = "DebugPanel" -- 7
function DebugPanel.prototype.____constructor(self) -- 7
	self.visible = false -- 8
	self.fpsAccum = 0 -- 10
	self.fpsFrames = 0 -- 11
	self.fps = 0 -- 12
end -- 7
function DebugPanel.prototype.toggle(self) -- 14
	self.visible = not self.visible -- 15
end -- 14
function DebugPanel.prototype.update(self, dt) -- 23
	self.fpsAccum = self.fpsAccum + dt -- 25
	self.fpsFrames = self.fpsFrames + 1 -- 26
	if self.fpsAccum >= 0.5 then -- 26
		self.fps = self.fpsFrames / self.fpsAccum -- 28
		self.fpsAccum = 0 -- 29
		self.fpsFrames = 0 -- 30
	end -- 30
	if Keyboard:isKeyDown("F3") then -- 30
		self:toggle() -- 33
	end -- 33
	if not self.visible then -- 33
		return -- 34
	end -- 34
	ImGui.Begin( -- 36
		"调试", -- 36
		function() -- 36
			ImGui.Text("FPS: " .. tostring(math.floor(self.fps))) -- 37
			ImGui.Text("相位: " .. ctx.phase) -- 38
			ImGui.Text("击杀: " .. tostring(ctx.stats.kills)) -- 39
			ImGui.Text("精英击杀: " .. tostring(ctx.stats.eliteKills)) -- 40
			ImGui.Text("Boss 击杀: " .. tostring(ctx.stats.bossKills)) -- 41
			ImGui.Text("波次: " .. tostring(ctx.stats.wave)) -- 42
			ImGui.Text("等级: " .. tostring(ctx.stats.playerLevel)) -- 43
			ImGui.Text(("存活: " .. tostring(math.floor(ctx.stats.timeAlive))) .. "s") -- 44
			local p = ctx.player -- 45
			if p ~= nil then -- 45
				ImGui.Text((("HP: " .. tostring(math.floor(p.hp))) .. "/") .. tostring(math.floor(p.maxHp))) -- 47
				ImGui.Text(((("位置: (" .. tostring(math.floor(p.pos.x))) .. ", ") .. tostring(math.floor(p.pos.y))) .. ")") -- 48
			end -- 48
		end -- 36
	) -- 36
end -- 23
__TS__SetDescriptor( -- 23
	DebugPanel.prototype, -- 23
	"isVisible", -- 23
	{get = function(self) -- 23
		return self.visible -- 19
	end}, -- 19
	true -- 19
) -- 19
____exports.debugPanel = __TS__New(____exports.DebugPanel) -- 55
return ____exports -- 55

File: game/debug/DebugPanel.ts
// 调试面板：ImGui 显示 FPS/击杀/波次/等级/相位；F3 切换
// Director 无 averageFPS，FPS 用平滑 1/dt 自算
import * as ImGui from 'ImGui';
import { KeyName, Keyboard } from 'Dora';
import { ctx } from 'game/core/GameContext';

export class DebugPanel {
	private visible = false;
	// FPS 平滑统计
	private fpsAccum = 0;
	private fpsFrames = 0;
	private fps = 0;

	toggle(): void {
		this.visible = !this.visible;
	}

	get isVisible(): boolean {
		return this.visible;
	}

	// 每帧调用：F3 开关 + 刷新面板内容
	update(dt: number): void {
		// FPS 统计（每 0.5s 更新一次平滑值）
		this.fpsAccum += dt;
		this.fpsFrames++;
		if (this.fpsAccum >= 0.5) {
			this.fps = this.fpsFrames / this.fpsAccum;
			this.fpsAccum = 0;
			this.fpsFrames = 0;
		}

		if (Keyboard.isKeyDown(KeyName.F3)) this.toggle();
		if (!this.visible) return;

		ImGui.Begin('调试', () => {
			ImGui.Text(`FPS: ${Math.floor(this.fps)}`);
			ImGui.Text(`相位: ${ctx.phase}`);
			ImGui.Text(`击杀: ${ctx.stats.kills}`);
			ImGui.Text(`精英击杀: ${ctx.stats.eliteKills}`);
			ImGui.Text(`Boss 击杀: ${ctx.stats.bossKills}`);
			ImGui.Text(`波次: ${ctx.stats.wave}`);
			ImGui.Text(`等级: ${ctx.stats.playerLevel}`);
			ImGui.Text(`存活: ${Math.floor(ctx.stats.timeAlive)}s`);
			const p = ctx.player;
			if (p !== undefined) {
				ImGui.Text(`HP: ${Math.floor(p.hp)}/${Math.floor(p.maxHp)}`);
				ImGui.Text(`位置: (${Math.floor(p.pos.x)}, ${Math.floor(p.pos.y)})`);
			}
		});
	}
}

// 单例导出：主控在 update 循环中调用 debugPanel.update(dt)
export const debugPanel = new DebugPanel();
